core/CPUTopology.cpp

```diff
@@ -395,6 +397,16 @@ void olympia::CoreTopologySimple::bindTree(sparta::RootTreeNode* root_node)
                 const std::string exe_pipe_out =
                     core_node + ".execute." + unit_name + ".ports.out_execute_pipe";
                 bind_ports(exe_pipe_in, exe_pipe_out);
+
+                for(uint32_t i = 0; i < pipelines[pipe_idx].size(); ++i){
+                    if(pipelines[pipe_idx][i] == "vset"){
+                        // only bind execute pipe -> decode port for an issue queue if it has a vset pipe
+                        const std::string exe_vset_out =
+                            core_node + ".execute." + unit_name + ".ports.out_vset";
+                        bind_ports(vset_in_decode, exe_vset_out);
+                        break; // break after because there should only be one vset per issue queue
```

Comment:
Should we have a check and assert instead of break to catch possible misconfiguration issues?

Reply:
Added sparta assertion checking for duplicate pipe definitions in [b8028a3](https://github.com/riscv-software-src/riscv-perf-model/pull/167/commits/b8028a38dd6f24d24687a0984feca2d5f382d1a9)